tracing/tracing.go

```diff
@@ -0,0 +1,121 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// propagator to use
+var propagator propagation.TextMapPropagator = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}) // traceparent header
+type Option interface {
+	apply(*options)
+}
+type options struct {
+	address string
+}
+
+// WithAddress takes an address in the form of Host:Port
+func WithAddress(addr string) Option {
+	return addrOption{address: addr}
+}
+
+type addrOption struct {
+	address string
+}
+
+func (o addrOption) apply(opts *options) {
+	opts.address = o.address
+}
+
+// OtlpGrpcExporter uses the otlptracegrpc modules and the otlptrace module to produce a new exporter at our default addr
+func OtlpGrpcExporter(ctx context.Context, opts ...Option) sdktrace.SpanExporter {
+
+	DefaultCollectorHost := "localhost"
+	var defaultCollectorPort uint16 = 4317
+	addr := fmt.Sprintf("%s:%d", DefaultCollectorHost, defaultCollectorPort)
+
+	options := options{
+		address: addr,
+	}
+
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	otlpClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithEndpoint(addr), //Not strictly needed as we use the defaults
+		otlptracegrpc.WithReconnectionPeriod(15*time.Second),
+		otlptracegrpc.WithInsecure(),
+	)
+
+	spanExporter, err := otlptrace.New(ctx, otlpClient)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Fatal logging in a lib is generally an anti-pattern. What about calling OtlpGrpcExporter makes it hard to handle a returned error?

Reply:
OtlpGrpcExporter is passed in as an WithExporter() option to the new() function of a wag client. 

Example:
`func createClient() (client *wot2.WagClient) {
	ctx := context.Background()
	client = wot2.New(ctx,
		"https://csm--wag-otel-test2--skylark.int.clever.com",
		// "http://localhost:8081",
		wot2.WithLogger(wot2.NewLogger("wagoteltest", wcl.Info)),

		wot2.WithInstrumentor(tracing.InstrumentedTransport),
		// wot2.WithExporter(tracing.JaegerExporter()),
		wot2.WithExporter(tracing.OtlpGrpcExporter(ctx)),
	)
	return

}`

Reply:
Totally open to suggestions here as I don't like the fatal log, but also don't want to complicate creating a client more than necessary.